modules/core.py

```diff
@@ -358,34 +354,51 @@ async def on_member_unban(self, guild: discord.Guild, user: discord.User):
                     self.bot, self.modLogs, 'unban', docID, reason, user=user, moderator=audited.user, public=True
                 )
 
-        embed = discord.Embed(color=discord.Color(0x88FF00), timestamp=datetime.datetime.utcnow())
-        embed.set_author(name=f'{user} ({user.id})', icon_url=user.avatar_url)
+        embed = discord.Embed(color=discord.Color(0x88FF00), timestamp=datetime.utcnow())
+        embed.set_author(name=f'{user} ({user.id})', icon_url=user.avatar.url)
         embed.add_field(name='Mention', value=f'<@{user.id}>')
 
         await self.serverLogs.send(':triangular_flag_on_post: User unbanned', embed=embed)
 
+    @commands.Cog.listener()
+    async def on_thread_join(self, thread):
+        # on_thread_join is called when a thread is created as well as joined
+        if not thread.me:
+            # We only want to send an API call if we aren't already in it
+            await thread.join()
+
     @commands.Cog.listener()
     async def on_message(self, message):
         if message.author.bot or message.webhook_id:
             return
 
-        if message.channel.type not in [discord.ChannelType.text, discord.ChannelType.news]:
+        if message.type != discord.MessageType.default:
+            return
+
+        if message.channel.type not in [
+            discord.ChannelType.text,
+            discord.ChannelType.news,
+            discord.ChannelType.public_thread,
+            discord.ChannelType.private_thread,
+        ]:
             logging.debug(f'Discarding non guild message {message.channel.type} {message.id}')
             return
 
         db = mclient.bowser.messages
         timestamp = int(time.time())
-        db.insert_one(
-            {
-                '_id': message.id,
-                'author': message.author.id,
-                'guild': message.guild.id,
-                'channel': message.channel.id,
-                'content': message.content,
-                'timestamp': timestamp,
-                'sanitized': False,
-            }
-        )
+        obj = {
+            '_id': message.id,
+            'author': message.author.id,
+            'guild': message.guild.id,
+            'channel': message.channel.id,
+            'content': message.content,
+            'timestamp': timestamp,
+            'sanitized': False,
+        }
+        if message.channel.type in [discord.ChannelType.public_thread, discord.ChannelType.private_thread]:
```

Comment:
I think it might be better to always include a parent_channel item, and get it to None for non-threads.

Reply:
I think you are right on that. I'll go ahead and do that as well as retroactively apply to old documents when merged